components/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/internal/provider/impl/WidgetInfoProviderImpl.java

```diff
@@ -0,0 +1,158 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.wso2.carbon.dashboards.core.internal.provider.impl;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.dashboards.core.widget.info.WidgetInfoProvider;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * This is the osgi service component to provide widget related information.
+ */
+@Component(
+        name = "org.wso2.carbon.dashboards.widget.WidgetInfoImpl",
+        service = WidgetInfoProvider.class,
+        immediate = true)
+public class WidgetInfoProviderImpl implements WidgetInfoProvider {
+
+    private static final Logger log = LoggerFactory.getLogger(WidgetInfoProviderImpl.class);
+    private static final String APP_PATH = "deployment/uufapps/";
+    private static final String CONF_FILE = "widgetConf.json";
```

Comment:
can we move this to a common constant class ... because there can be more constants coming in on next development rotations ..